waspc/src/Wasp/AppSpec/Valid.hs

```diff
@@ -49,6 +56,43 @@ validateExactlyOneAppExists spec =
         GenericValidationError $
           "You have more than one 'app' declaration in your Wasp app. You have " ++ show (length apps) ++ "."
 
+validateWasp :: AppSpec -> [ValidationError]
+validateWasp = validateWaspVersion . Wasp.version . App.wasp . snd . getApp
+
+validateWaspVersion :: String -> [ValidationError]
+validateWaspVersion specWaspVersionStr = eitherUnitToErrorList $ do
+  specWaspVersionRange <- parseWaspVersionRange specWaspVersionStr
+  unless
+    ( SV.isVersionInRange WV.waspVersion specWaspVersionRange
+    )
+    $ Left $ incompatibleVersionError WV.waspVersion specWaspVersionRange
+  where
+    -- TODO: Use version range parser from SemanticVersion when it is fully implemented.
+
+    parseWaspVersionRange :: String -> Either ValidationError SV.Range
+    parseWaspVersionRange waspVersionRangeStr = do
+      let (_ :: String, _ :: String, _ :: String, waspVersionRangeDigits :: [String]) =
+            waspVersionRangeStr =~ ("\\`\\^([0-9]+)\\.([0-9]+)\\.([0-9]+)\\'" :: String)
+
+      waspSpecVersion <- case mapM readMaybe waspVersionRangeDigits of
+        Just [major, minor, patch] -> Right $ SV.Version major minor patch
```

Comment:
Please put a comment here explaining that we use only `^` for now only because it was the easiest solution to start with and that we plan to allow any semver definition in the future.